exporter/awsemfexporter/datapoint.go

```diff
@@ -0,0 +1,326 @@
+type IntDataPointSlice struct {
+	instrumentationLibraryName string
+	rateCalculationMetadata
+	pdata.IntDataPointSlice
+}
+
+// DoubleDataPointSlice is a wrapper for pdata.DoubleDataPointSlice
+type DoubleDataPointSlice struct {
+	instrumentationLibraryName string
+	rateCalculationMetadata
+	pdata.DoubleDataPointSlice
+}
+
+// DoubleHistogramDataPointSlice is a wrapper for pdata.DoubleHistogramDataPointSlice
+type DoubleHistogramDataPointSlice struct {
+	instrumentationLibraryName string
+	pdata.DoubleHistogramDataPointSlice
+}
+
+// DoubleSummaryDataPointSlice is a wrapper for pdata.DoubleSummaryDataPointSlice
+type DoubleSummaryDataPointSlice struct {
+	instrumentationLibraryName string
+	pdata.DoubleSummaryDataPointSlice
+}
+
+// At retrieves the IntDataPoint at the given index and performs rate calculation if necessary.
+func (dps IntDataPointSlice) At(i int) DataPoint {
+	metric := dps.IntDataPointSlice.At(i)
+	labels := createLabels(metric.LabelsMap(), dps.instrumentationLibraryName)
+	timestamp := unixNanoToMilliseconds(metric.Timestamp())
+
+	var metricVal interface{}
+	metricVal = metric.Value()
+	if dps.needsCalculateRate {
+		rateKey := createMetricKey(labels, dps.rateKeyParams)
+		rateTS := dps.timestamp
+		if timestamp > 0 {
+			// Use metric timestamp if available
+			rateTS = timestamp
+		}
+		metricVal = calculateRate(rateKey, metricVal, rateTS)
+	}
+
+	return DataPoint{
+		Value:     metricVal,
+		Labels:    labels,
+		Timestamp: timestamp,
+	}
+}
+
+// At retrieves the DoubleDataPoint at the given index and performs rate calculation if necessary.
+func (dps DoubleDataPointSlice) At(i int) DataPoint {
+	metric := dps.DoubleDataPointSlice.At(i)
+	labels := createLabels(metric.LabelsMap(), dps.instrumentationLibraryName)
+	timestamp := unixNanoToMilliseconds(metric.Timestamp())
+
+	var metricVal interface{}
+	metricVal = metric.Value()
+	if dps.needsCalculateRate {
+		rateKey := createMetricKey(labels, dps.rateKeyParams)
+		rateTS := dps.timestamp
+		if timestamp > 0 {
+			// Use metric timestamp if available
+			rateTS = timestamp
+		}
+		metricVal = calculateRate(rateKey, metricVal, rateTS)
+	}
+
+	return DataPoint{
+		Value:     metricVal,
+		Labels:    labels,
+		Timestamp: timestamp,
+	}
+}
+
+// At retrieves the DoubleHistogramDataPoint at the given index.
+func (dps DoubleHistogramDataPointSlice) At(i int) DataPoint {
+	metric := dps.DoubleHistogramDataPointSlice.At(i)
+	labels := createLabels(metric.LabelsMap(), dps.instrumentationLibraryName)
+	timestamp := unixNanoToMilliseconds(metric.Timestamp())
+
+	var minBound, maxBound float64
+	bucketBounds := metric.ExplicitBounds()
+	if len(bucketBounds) > 0 {
+		minBound = bucketBounds[0]
+		maxBound = bucketBounds[len(bucketBounds)-1]
+	}
+
+	return DataPoint{
+		Value: &CWMetricStats{
+			Min:   minBound,
+			Max:   maxBound,
+			Count: metric.Count(),
+			Sum:   metric.Sum(),
+		},
+		Labels:    labels,
+		Timestamp: timestamp,
```

Comment:
The code here that handles `DoubleHistogramDataPointSlice` is not the latest change, please  check out here https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/master/exporter/awsemfexporter/metric_translator.go#L341

Reply:
Yea, that was what we had initially, but I added this in because I thought we might want this in the metric. Should I remove it?

Reply:
Yes, we should remove it. The minBound & maxBound doesn't fit our Min and Max values in CW Metrics semantically. If we have it here, it will display the wrong values for customers.